ipaserver/plugins/dns.py

```diff
@@ -2431,15 +2431,14 @@ class dnszone(DNSZoneBase):
             normalizer=normalize_zonemgr,
             autofill=True,
         ),
-        Int('idnssoaserial',
+        Int('idnssoaserial?',
             # Deprecated
             cli_name='serial',
             label=_('SOA serial'),
             doc=_('SOA record serial number'),
             minvalue=1,
             maxvalue=4294967295,
-            default_from=_create_zone_serial,
-            autofill=True,
+            deprecated=True,
```

Comment:
This not only deprecates the option but makes it not possible for a user to provide it, either on the CLI or the UI. Is that intentional?

Reply:
Yes, that is the main point behind this fix (aside from not showing the warning message anymore). 

As the SOA serial is managed automatically via autoincrement, the idea is that the user couldn't provide the SOA serial parameter manually (via CLI or WebUI). Also, we prevent it from being generated via autofill, which naturally avoids the warning.

Reply:
As for the CLI, the option was already not available for the user to set SOA serial. The issue was that the API was still accepting it.